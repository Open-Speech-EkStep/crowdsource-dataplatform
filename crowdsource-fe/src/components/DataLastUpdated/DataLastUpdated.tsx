import moment from 'moment';
import { useTranslation } from 'next-i18next';

import apiPaths from 'constants/apiPaths';
import useFetch from 'hooks/useFetch';
import type { DataLastUpdated as DataLastUpdatedType } from 'types/DataLastUpdated';

function DataLastUpdated() {
  const { t } = useTranslation();

  const { data } = useFetch<Array<DataLastUpdatedType>>(apiPaths.lastUpdatedTime);

  if (data) {
    const lastUpdatedAt = moment(data[0]['timezone']).format('DD-MM-YYYY, h:mm:ss a');
    return (
      <div>
        {t('dataLastUpdated')}: {lastUpdatedAt}
      </div>
    );
  } else {
    return <div>Hello</div>;
  }
}

export default DataLastUpdated;
