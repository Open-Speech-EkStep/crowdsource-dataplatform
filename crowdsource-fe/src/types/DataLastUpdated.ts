export interface DataLastUpdated {
  timezone: string;
}
