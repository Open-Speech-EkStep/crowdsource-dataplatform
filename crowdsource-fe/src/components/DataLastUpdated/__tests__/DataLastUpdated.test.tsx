import { render, waitFor } from '@testing-library/react';

import DataLastUpdated from '../DataLastUpdated';

describe('DataLastUpdated', () => {
  const setup = async () => {
    const url = '/aggregated-json/lastUpdatedAtQuery.json';
    fetchMock
      .doMockOnceIf(url)
      .mockResponseOnce(JSON.stringify([{ timezone: '2021-10-11T13:30:05.744813' }]));

    const renderResult = render(<DataLastUpdated />);

    await waitFor(() => {
      expect(fetchMock).toHaveBeenCalledWith(url);
    });
    return renderResult;
  };

  it('should render snapshot', async () => {
    const { asFragment } = await setup();

    expect(asFragment()).toMatchSnapshot();
  });
});
