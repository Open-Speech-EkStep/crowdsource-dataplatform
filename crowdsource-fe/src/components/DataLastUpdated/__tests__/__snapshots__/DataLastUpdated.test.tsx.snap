// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataLastUpdated should render snapshot 1`] = `
<DocumentFragment>
  <div>
    dataLastUpdated: 11-10-2021, 1:30:05 pm
  </div>
</DocumentFragment>
`;
